### PR TITLE
Fixups and convert to anyhow

### PR DESCRIPTION
**Pull request by Morganamilo**
I went along with the anyhow thing to try and convince you :P Error messages are now much more readable.

If you don't like it I'll just pop the last commit.

before: `Error: Alpm(NotADir)`

after: `error: failed to initialize alpm: root='/' dbpath='fcedew': could not find or read directory`

### CODE REVIEW

File: rust/aura/src/main.rs [hunk 1]
```diff
@@ -32,14 +32,12 @@ fn main() -> Result<(), Error> {
     // Establish common file paths.
     let logp: &Path = args
         .logfile
-        .as_ref()
-        .map(|p| p.as_path())
+        .as_deref()
```
Comment (fosskers) [inline rust/aura/src/main.rs:68]: I will remember this.

**Comment by fosskers** [comment]
I may need to forego the `anyhow` usage for now, considering that I will want all user-facing messages to be localized. Another reason for having the explicit variant wrappers around each error type. 

I can merge this and then hand-write the `Error` trait conversion code so that things are easier for you in the meantime.

**Comment by fosskers** [comment]
And double-checking the way docs render with the lib now gone, everything looks as expected :+1: I frequently utilize my own projects' docs while developing so that was critical to me.

**Comment by fosskers** [comment]
The commits are now included back in the `colin/rust` branch, although Github seems not to have detected that here, so I'll close manually.

File: rust/aura/src/main.rs (hunk 1 above)
Comment (ratijas) [reply rust/aura/src/main.rs:68]: Oh, come on, just throw `&` and `*` at it — as many as needed, until it compiles.

**Pull request closed without merging**